tests/epyccel/test_arrays.py

```diff
@@ -1495,6 +1511,1171 @@ def test_array_variable_size():
+
+def test_array_1d_slice_1():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_1
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_2():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_2
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_3():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_3
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_4():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_4
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_5():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_5
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_6():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_6
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_7():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_7
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_8():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_8
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_9():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_9
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_10():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_10
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_11():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_11
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+def test_array_1d_slice_12():
+    a = np.empty(21, dtype=np.int)
+    for i in range(21):
+        a[i] = 2**i
+    f1 = arrays.array_1d_slice_12
+    f2 = epyccel(f1)
+
+    assert np.array_equal(f1(a), f2(a))
+
+#==============================================================================
+# TEST : 2d array slices order F
+#==============================================================================
+
+def test_array_2d_F_slice_1():
+    a = np.empty((21,21), dtype=np.int, order = 'F')
+    for i in range(21):
+        for j in range(21):
+            a[i][j] = 2**j
```

Comment:
The standard way to index an N-dimensional Numpy array is with N indices, so you should write `a[i, j] = 2**j`. Here you have instead:
- `a`: 2D array
- `a[i]`: 1D slicing of `a`
- `a[i][j]`: get element `j` of `a[i]`

which is definitely slower.

Anyway, I suspect that the fastest way to fill this 2D array is the following:
```python
with np.nditer(a, op_flags=['readwrite']) as it:
    for x in it:
        x[...] = 2**it.multi_index[1]
```

But without going into the Numpy craziness above, I think we could simply make `a` a module variable given that it is never modified by the various functions. So we could define, for example:
```python
a_1d   = ...
a_2d_c = ...
a_2d_f = ...
```

Please have a look at these approaches and time them, and let me know what you find out.

Reply:
done. generating an array in a module and use it for all tests is faster.